File: C#/Integer Break/Integer Break.cs
public class Solution {
    public int IntegerBreak(int n) {
        int[] dp=new int[n+1];
        for(int i=2;i<=n;i++){
            int curMax=0;
            for(int j=1;j<i;j++){
                curMax=Math.Max(dp[i-j]*j,(i-j)*j);
                if(dp[i]<curMax) dp[i]=curMax;
            }
        }
        return dp[n];
    }
}

File: Python/Integer Break/Integer Break.py
class Solution:
    def integerBreak(self, n: int) -> int:
        dp=[0]*59
        for i in range(2,59):
            for j in range(1,i):
                dp[i]=max(dp[i],j*dp[i-j],j*(i-j))
        return dp[n]
